File: 1822-sign-of-the-product-of-an-array/1822-sign-of-the-product-of-an-array.cpp
class Solution {
public:
    int arraySign(vector<int>& nums) {
        if(nums[0] == 0)
            return 0;
        int product = nums[0];
        for(int i = 1; i<nums.size(); i++)
        {
            if(nums[i] == 0)
                return 0;
            else
                if(nums[i] > 0)
                    product *= 1;
                else
                    product *= -1;
        }
        if(product > 0)
            return 1;
        else
            return -1;
    }
};

File: 1822-sign-of-the-product-of-an-array/1822-sign-of-the-product-of-an-array.py
class Solution:
# simple solution by looping and checking for positive/negatives
    def arraySign(self, nums: List[int]) -> int:
        if(nums[0] == 0):
            return 0
        product = nums[0]
        nums = nums[1::]
        for i in nums:
            if i == 0:
                return 0
            product *= i;
        if product > 0:
            return 1
        else:
            return -1
        